### ISSUE TITLE
Content lib - Modify-method changes values to strings in the site config

### ISSUE THREAD

**Issue by arneroen** [enonic/xp#7255]
Hi,

I think I've found a bug where contentLib.modify will change booleans in the siteConfig to strings (eg true -> "true").
(Enonic XP version 7.0.1)
To reproduce it:

```
<?xml version="1.0" encoding="UTF-8"?>
<site>
  <form>
    <input type="CheckBox" name="someConfigValue">
      <label>label</label>
      <occurrences minimum="0" maximum="1"/>
      <default>checked</default>
    </input>
  </form>
</site>

```

And the function that modifies it is just
```
  contentLib.modify({
    key: '/path',
    branch: 'draft',
    requireValid: false, //to see the changes in the data structures. 
    editor: site => {
      return site
    }
  })
```

Using data toolbox we can inspect the config values before and after
Before:
```
"data": {
  "siteConfig": {
    "applicationKey": "com.example",
    "config": {
      "someConfigValue": true
    }
  }
}
```
After: 
```
"data": {
  "siteConfig": {
    "applicationKey": "com.example",
    "config": {
      "someConfigValue": "true"
    }
  }
}
```

**Comment by GlennRicaud** [comment]
@arneroen Thank you for reporting this bug. We will fix it in 7.0.3

**Comment by GlennRicaud** [comment]
@vbradnitski

The current fix is good and should be included in the final version.

But it does not solve the wider problem:
SiteConfig forms are not used when modifying contents...
One solution would be to inline the site forms similar to mixins when calling FormJsonToPropertyTreeTranslator

To test the solution try the same thing but with the following site config

```
<?xml version="1.0" encoding="UTF-8"?>
<site>
  <form>
    <input name="myDate" type="Date">
      <label>Date</label>
      <occurrences minimum="0" maximum="0"/>
    </input>
  </form>
</site>
```
